pkg/data/message.go

```diff
@@ -0,0 +1,132 @@
+package data
+
+import (
+	"fmt"
+	"time"
+)
+
+// ErrorMessageNotFound : Message specific errors
+var ErrorMessageNotFound = fmt.Errorf("Message not found")
+
+// Message defines the structure for an API message.
+// Formatting done with json tags to the right. "-" : don't include when encoding to json
+type Message struct {
+	ID             int    `json:"id"`
+	UserID         int    `json:"userid"`
+	ConversationID int    `json:"conversationid"`
+	Text           string `json:"text" validate:"required"`
+	CreatedOn      string `json:"-"`
+	UpdatedOn      string `json:"-"`
+	DeletedOn      string `json:"-"`
+}
+
+// Messages is a collection of Message
+type Messages []*Message
+
+// All of these functions will become database calls in the future
+
+// GetMessages returns the list of messages
+func GetMessages() Messages {
+	return messageList
+}
+
+// GetMessageByID returns a single message with the given id
+func GetMessageByID(id int) (*Message, error) {
+	index := findIndexByMessageID(id)
+	if index == -1 {
+		return nil, ErrorMessageNotFound
+	}
+	return messageList[index], nil
+}
+
+// GetMessageByConversationID returns an array of messages corresponding to the conversation
+func GetMessagesByConversationID(id int) ([]*Message, error) {
+	messages := []*Message{}
+	for _, v := range messageList {
+		if v.ConversationID == id {
+			messages = append(messages, v)
+		}
+	}
+	if len(messages) <= 0 {
+		return nil, ErrorMessageNotFound
+	}
+	return messages, nil
+}
+
+// AddMessage creates a new message
+func AddMessage(message *Message) error {
+	_, err := GetConversationByID(message.ConversationID)
```

Comment:
Est-ce que la validation du ConversationID serait vieux comme un custom validator dans la struct?

Reply:
Validators should only check if the structure of the json is correct. Checking that the conversation exists should be done here

Reply:
We don't want to make a db call in the validators

Reply:
Avant, je validais avec le middleware, mais je l'ai mis comme ça après le commentaire de Malcolm sur le fait qu'on valide seulement la structure du JSON dans le middleware